plugins/osdn/common.go

```diff
@@ -140,64 +140,64 @@ func (oc *OsdnController) validateNetworkConfig(clusterNetwork, serviceNetwork *
 		return err
 	}
 	for _, svc := range services {
-		if !serviceNetwork.Contains(net.ParseIP(svc.Spec.ClusterIP)) {
-			errList = append(errList, fmt.Errorf("Error: Existing service with IP: %s is not part of service network: %s", svc.Spec.ClusterIP, serviceNetwork.String()))
+		if !ni.ServiceNetwork.Contains(net.ParseIP(svc.Spec.ClusterIP)) {
+			errList = append(errList, fmt.Errorf("Error: Existing service with IP: %s is not part of service network: %s", svc.Spec.ClusterIP, ni.ServiceNetwork.String()))
 		}
 	}
 
 	return kerrors.NewAggregate(errList)
 }
 
-func (oc *OsdnController) isClusterNetworkChanged(clusterNetworkCIDR string, hostBitsPerSubnet int, serviceNetworkCIDR string) (bool, error) {
-	clusterNetwork, hostSubnetLength, serviceNetwork, err := oc.Registry.GetNetworkInfo()
+func (oc *OsdnController) isClusterNetworkChanged(curNetwork *NetworkInfo) (bool, error) {
+	oldNetwork, err := oc.Registry.GetNetworkInfo()
 	if err != nil {
 		return false, err
 	}
-	if clusterNetworkCIDR != clusterNetwork.String() ||
-		hostSubnetLength != hostBitsPerSubnet ||
-		serviceNetworkCIDR != serviceNetwork.String() {
+
+	if curNetwork.ClusterNetwork.String() != oldNetwork.ClusterNetwork.String() ||
+		curNetwork.HostSubnetLength != oldNetwork.HostSubnetLength ||
+		curNetwork.ServiceNetwork.String() != oldNetwork.ServiceNetwork.String() {
 		return true, nil
 	}
 	return false, nil
 }
 
 func (oc *OsdnController) StartMaster(clusterNetworkCIDR string, clusterBitsPerSubnet uint, serviceNetworkCIDR string) error {
 	// Validate command-line/config parameters
-	hostBitsPerSubnet := int(clusterBitsPerSubnet)
-	clusterNetwork, _, serviceNetwork, err := ValidateClusterNetwork(clusterNetworkCIDR, hostBitsPerSubnet, serviceNetworkCIDR)
+	network, err := ValidateClusterNetwork(clusterNetworkCIDR, int(clusterBitsPerSubnet), serviceNetworkCIDR)
```

Comment:
Maybe 'network' should be 'ni' here to be consistent with other parts of the code?  I won't block a merge for that though.